# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLAbsPositionEditor.cpp
# Commit: 9ee7acab98a4
# Full Hash: 9ee7acab98a470b2860125ba3deb439a64dc83e7
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 1
# Description:
#   Bug 1529884 - part 6: Through subject principal at Document::ExecCommand() to constructor of EditorBase::AutoEditActionDataSetter r=smaug
#   
#   `Document::ExecCommand()` knows subject principal.  This patch makes it tell
#   `EditorCommand::DoCommand()` and `EditorCommand::DoCommandParam()`.  Then,
#   makes they tell each editor public methods which may cause dispatching
# ==============================================================================

diff -r 258c4da663ca -r 9ee7acab98a4 editor/libeditor/HTMLAbsPositionEditor.cpp
--- a/editor/libeditor/HTMLAbsPositionEditor.cpp	Mon Jun 10 10:26:04 2019 +0000
+++ b/editor/libeditor/HTMLAbsPositionEditor.cpp	Mon Jun 10 10:27:07 2019 +0000
@@ -31,6 +31,7 @@
 #include "nsIDOMWindow.h"
 #include "nsIHTMLObjectResizer.h"
 #include "nsINode.h"
+#include "nsIPrincipal.h"
 #include "nsISupportsImpl.h"
 #include "nsISupportsUtils.h"
 #include "nsLiteralString.h"
@@ -44,11 +45,12 @@
 
 using namespace dom;
 
-nsresult HTMLEditor::SetSelectionToAbsoluteOrStatic(bool aEnabled) {
+nsresult HTMLEditor::SetSelectionToAbsoluteOrStaticAsAction(
+    bool aEnabled, nsIPrincipal* aPrincipal) {
   MOZ_ASSERT(IsEditActionDataAvailable());
 
   AutoEditActionDataSetter editActionData(
-      *this, EditAction::eSetPositionToAbsoluteOrStatic);
+      *this, EditAction::eSetPositionToAbsoluteOrStatic, aPrincipal);
   if (NS_WARN_IF(!editActionData.CanHandle())) {
     return NS_ERROR_NOT_INITIALIZED;
   }
@@ -142,11 +144,12 @@
   mCSSEditUtils->SetCSSProperty(aElement, *nsGkAtoms::z_index, zIndexStr);
 }
 
-nsresult HTMLEditor::AddZIndex(int32_t aChange) {
+nsresult HTMLEditor::AddZIndexAsAction(int32_t aChange,
+                                       nsIPrincipal* aPrincipal) {
   MOZ_ASSERT(IsEditActionDataAvailable());
 
   AutoEditActionDataSetter editActionData(
-      *this, EditAction::eIncreaseOrDecreaseZIndex);
+      *this, EditAction::eIncreaseOrDecreaseZIndex, aPrincipal);
   if (NS_WARN_IF(!editActionData.CanHandle())) {
     return NS_ERROR_NOT_INITIALIZED;
   }